modules/editor-service/src/main/java/org/opencastproject/editor/EditorServiceImpl.java

```diff
@@ -0,0 +1,779 @@
+        for (SmilMediaObject video : mediaContainer.getElements()) {
+          if (video instanceof SmilMediaElement) {
+            SmilMediaElement videoElem = (SmilMediaElement) video;
+            try {
+              // pick longest element
+              if (tuple == null || (videoElem.getClipEndMS()
+                      - videoElem.getClipBeginMS()) > (Long) tuple.getEnd() - (Long) tuple.getStart()) {
+                tuple = new SegmentData(videoElem.getClipBeginMS(), videoElem.getClipEndMS());
+              }
+            } catch (SmilException e) {
+              logger.warn("Media element '{}' of SMIL catalog '{}' seems to be invalid: {}",
+                      videoElem, smil, e);
+            }
+          }
+        }
+        if (tuple != null) {
+          segments.add(tuple);
+        }
+      }
+    }
+    return segments;
+  }
+
+  @Override
+  public EditingData getEditData(final String mediaPackageId) throws EditorServiceException {
+    // Select tracks
+    Opt<Event> eventOpt = getEvent(mediaPackageId);
+    if (eventOpt.isNone()) {
+      errorExit("Not Found", mediaPackageId, ErrorStatus.MEDIAPACKAGE_NOT_FOUND);
+    }
+    final Event event = eventOpt.get();
+
+    MediaPackage mp = null;
+    try {
+      mp = index.getEventMediapackage(event);
+    } catch (IndexServiceException e) {
+      errorExit("Not Found", mediaPackageId, ErrorStatus.MEDIAPACKAGE_NOT_FOUND);
+    }
+
+    final Opt<Publication> internalPubOpt = getInternalPublication(mp);
+    if (internalPubOpt.isNone() || internalPubOpt.isEmpty()) {
+      errorExit("No internal publication", mediaPackageId, ErrorStatus.NO_INTERNAL_PUBLICATION);
+    }
+    Publication internalPub = internalPubOpt.get();
+
+    // Get existing segments
+    List<SegmentData> segments = getSegments(mp);
+    segments.addAll(getDeletedSegments(mp, segments));
+    sortSegments(segments);
+
+
+    // Get workflows
+    List<WorkflowData> workflows = new ArrayList<>();
+    for (WorkflowDefinition workflow : getEditingWorkflows()) {
+      workflows.add(new WorkflowData(workflow.getId(), workflow.getTitle(), workflow.getDisplayOrder(),
+              workflow.getDescription()));
+    }
+
+    final Map<String, String> latestWfProperties = WorkflowPropertiesUtil
+            .getLatestWorkflowProperties(assetManager, mediaPackageId);
+    // The properties have the format "hide_flavor_audio" or "hide_flavor_video", where flavor is preconfigured.
+    // We filter all the properties that have this format, and then those which have values "true".
+    final Collection<Tuple<String, String>> hiddens = latestWfProperties.entrySet()
+            .stream()
+            .map(p -> tuple(p.getKey().split("_"), p.getValue()))
+            .filter(p -> p.getA().length == 3)
+            .filter(p -> p.getA()[0].equals("hide"))
+            .filter(p -> p.getB().equals("true"))
+            .map(p -> tuple(p.getA()[1], p.getA()[2]))
+            .collect(Collectors.toSet());
```

Comment:
The problem by using a `Map<String, String>` is to find a unique key that will not be used twice in the workflow properties. 
The suggested way to use `hide` as a key makes not so much sense in my opinion as the whole map only consists of hidden elements and it leads to an exception if audio and video are configured to be hidden (even if this makes no sense). I can't use `audio` or `video` as key because there might be multiple flavors configured. 
I can't use the flavor as key because it leads to an exception if audio and video are configured to be hidden for the same flavor.

Reply:
That makes sense. But if `p[0]` is always `hide` we don't actually need to store it at all, and we could just collect to a set/list containing `p[2]`. And I say all of this, not having had a look at the current code. So ignore it for now, I'll confirm or contradict myself later :grinning:

Reply:
Okay, my `p[0]` in the collector never made sense. It should have been a `p[1]`. But the current solution seems reasonable if we can have duplicated keys.